### PR TITLE
docs: allow having Internal section with stories only visible for dev mode

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/6437258196

image

### CODE REVIEW

File: packages/core/.storybook/manager.jsx [hunk 1]
```diff
@@ -24,6 +24,13 @@ addons.setConfig({
...
       return <SidebarItem status={parameters.status}>{name.replace(storyStatus, "").trim()}</SidebarItem>;
+    },
+    filters: {
+      patterns: filterInternalStoryDocsPage
     }
   }
 });
+
+function filterInternalStoryDocsPage(item) {
+  return process.env.STORYBOOK_ENV === "development" || !item.tags?.includes?.("internal");
```
Comment (talkor) [inline packages/core/.storybook/manager.jsx:35]: Can't we filter the entire section without the tags?

Comment (talkor) [inline packages/core/.storybook/manager.jsx:35]: Storybook uses env variables, I think you can get them somehow here

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: I've tried to also use their support, and their docs, couldn't find anything related

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: basically, we can say that if the name includes Internal.
I could add it as an extra check, and leave the current, just for cases where if we would like to have an internal story inside an existing category. wdyt?

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: would love to know if you find something, I'll change it 🙏🏼

Comment (talkor) [reply packages/core/.storybook/manager.jsx:35]: You can use `process.env.NODE_ENV`

Comment (talkor) [reply packages/core/.storybook/manager.jsx:35]: It should work also with internals stories, no? Anyway I think it's fine for now with tags. It's a new feature in SB7 no? We can use them for the beta/deprecated tags as well

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: Are they setting it?

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: - yep it is
- you mean adding styled tags?
- and if I remove the check for `tags` and only exclude the `Internal` category, then it won't exclude internal stories that are inside regular (exported) categories

Comment (talkor) [reply packages/core/.storybook/manager.jsx:35]: Yes tried it locally also, and they use `production` for build

Comment (talkor) [reply packages/core/.storybook/manager.jsx:35]: No I mean the sidebar tags, we use the name to decide whether a story is a beta or deprecated, I think using tags is better for that purpose. Let's stay with tags, forget about what I said 😅

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: what do they supply in `dev`?
I'm not sure if we're exporting as production build

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: ohhh I got you
TechDebt task 🙃

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:35]: https://monday.monday.com/boards/3532714909/pulses/6453084060

File: packages/core/.storybook/manager.jsx [hunk 2]
```diff
@@ -24,6 +25,16 @@ addons.setConfig({
...
       return <SidebarItem status={parameters.status}>{name.replace(storyStatus, "").trim()}</SidebarItem>;
+    },
+    filters: {
+      patterns: filterStory
     }
   }
 });
+
+function filterStory(item) {
+  console.log(process.env.NODE_ENV === "development", item.title, ...item.tags, { item });
```
Comment (talkor) [inline packages/core/.storybook/manager.jsx:36]: lol

**Review by talkor — APPROVED** [review]
Awesome, just remove the console

File: packages/core/.storybook/manager.jsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:36]: 🙃

Comment (YossiSaadi) [reply packages/core/.storybook/manager.jsx:36]: Thanks!

**Pull request merged**